arrow/arrow.py

```diff
@@ -1032,81 +1148,69 @@ def humanize(
-                    delta = sign * delta / self._SECS_PER_MONTH
+                    delta = sign * delta_second / self._SECS_PER_MONTH
                 elif granularity == "year":
-                    delta = sign * delta / self._SECS_PER_YEAR
+                    delta = sign * delta_second / self._SECS_PER_YEAR
                 else:
                     raise AttributeError(
                         "Invalid level of granularity. Please select between 'second', 'minute', 'hour', 'day', 'week', 'month' or 'year'"
                     )
 
                 if trunc(abs(delta)) != 1:
-                    granularity += "s"
+                    granularity += "s"  # type: ignore
                 return locale.describe(granularity, delta, only_distance=only_distance)
 
             else:
-                timeframes = []
-                if "year" in granularity:
-                    years = sign * delta / self._SECS_PER_YEAR
-                    delta %= self._SECS_PER_YEAR
-                    timeframes.append(["year", years])
-
-                if "month" in granularity:
-                    months = sign * delta / self._SECS_PER_MONTH
-                    delta %= self._SECS_PER_MONTH
-                    timeframes.append(["month", months])
-
-                if "week" in granularity:
-                    weeks = sign * delta / self._SECS_PER_WEEK
-                    delta %= self._SECS_PER_WEEK
-                    timeframes.append(["week", weeks])
-
-                if "day" in granularity:
-                    days = sign * delta / self._SECS_PER_DAY
-                    delta %= self._SECS_PER_DAY
-                    timeframes.append(["day", days])
-
-                if "hour" in granularity:
-                    hours = sign * delta / self._SECS_PER_HOUR
-                    delta %= self._SECS_PER_HOUR
-                    timeframes.append(["hour", hours])
-
-                if "minute" in granularity:
-                    minutes = sign * delta / self._SECS_PER_MINUTE
-                    delta %= self._SECS_PER_MINUTE
-                    timeframes.append(["minute", minutes])
-
-                if "second" in granularity:
-                    seconds = sign * delta
-                    timeframes.append(["second", seconds])
+                timeframes: List[Tuple[TimeFrames, float]] = []
+
+                def gather_timeframes(_delta: float, _frame: TimeFrames) -> float:
+                    if _frame in granularity:
+                        value = sign * _delta / self._SECS_MAP[_frame]
+                        _delta %= self._SECS_MAP[_frame]
+                        if trunc(abs(value)) != 1:
+                            timeframes.append((cast(TimeFrames, _frame + "s"), value))
+                        else:
+                            timeframes.append((_frame, value))
+                    return _delta
+
+                delta = float(delta_second)
+                frames: Tuple[TimeFrames, ...] = (
```

Comment:
Does the `...` syntax here mean we are only using a subset of `TimeFrames`? Sorry I am new to mypy/typing syntax hah!

Reply:
Oh nevermind... this seems to just be spreading the same type over the length of the tuple!